src/file.ts

```diff
@@ -2281,25 +2285,76 @@ class File extends ServiceObject<File> {
       throw new Error('The expiration date provided was invalid.');
     }
 
-    if (expiresInMSeconds < Date.now()) {
+    if (expiresInMSeconds < this.getDate().valueOf()) {
       throw new Error('An expiration date cannot be in the past.');
     }
 
     const expiresInSeconds =
         Math.round(expiresInMSeconds / 1000);  // The API expects seconds.
 
-    const config: GetSignedUrlConfigInternal = Object.assign({}, cfg);
-
-    config.action = ({
+    const verb = ({
       read: 'GET',
       write: 'PUT',
       delete: 'DELETE',
       resumable: 'POST',
-    } as {[index: string]: string})[config.action];
```

Comment:
I think I saw a conformance test looking for this type somewhere in another PR lol